the-collab-lab/tcl-35-smart-shopping-list
# Aa fo connect to firestore

## Description
This PR reads from and writes to the Firestore database so that users can persist information. In this PR, Ayu and Fatima created two components for the feature(AddItem and ListItem). Ayu worked on the ListItem component and was able to read from firestore while Fatima worked on the AddItem component to write to firestore. The code was then refactored and clean up. We learnt how to work together and collaborate effectively, we also learnt how to debug and fix errors by googling and asking questions.

### Write items to the database
We set up firestore on our codebase and connected it to the database so that we can write to it. We created a button and wrote a function for the button that helps submit our hard-coded data to firestore on a click action.

Check out more documentation and examples of how to write to firestore [here](https://firebase.google.com/docs/firestore/manage-data/add-data)

### Read items to the database
We set up firestore on our codebase and connected it to the database so that we can read from it. We populated our data from firestore to users immediately after the page loads. We make use of useeffect hooks to create a function that reads from firestore.

More documentation of effect hooks [here](https://reactjs.org/docs/hooks-effect.html)
Check out more documentation and examples of how to read from firestore [here](https://firebase.google.com/docs/firestore/query-data/get-data)



## Related Issue


closes #1 

## Acceptance Criteria

  ✓ The following have been added as project dependencies: firebase v9
  ✓Make a change in the Firestore database and it shows up in the app
  ✓Make a change in the app and it shows up in the Firestore database



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓| :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|   ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After


![collabpr](https://user-images.githubusercontent.com/57072944/136308639-2fd24214-8e75-4d1a-b277-362eb1365f5e.png)

## Testing Steps / QA Criteria



- From your terminal, pull down this branch with git pull origin aa-fo-connect-to-firestore and check that branch out with git checkout aa-fo-connect-to-firestore
- Then npm update to install the newly added dependencies locally and npm start to launch the app.
- Onload, the data in firestore is populated and displayed on the page. 
- To test writing to firestore feature, navigate to AddItem.js file and input a new itemName and buyingTime, save your changes, and click Add Item button on the page to add the item. The item is added to firestore and displayed on the page.
